src/cmap/handshake/client_metadata.ts

```diff
@@ -152,8 +153,60 @@ export function makeClientMetadata(options: MakeClientMetadataOptions): ClientMe
       }
     }
   }
+  return metadataDocument.toObject() as ClientMetadata;
+}
+
+let isDocker: boolean;
+let dockerPromise: Promise<boolean>;
+/** @internal */
+async function getContainerMetadata() {
+  const containerMetadata: Record<string, any> = {};
+  if (isDocker == null) {
+    dockerPromise ??= fs.access('/.dockerenv').then(
+      () => true,
+      () => false
+    );
+    isDocker = await dockerPromise;
+  }
+
```

Comment:
(optional) the original suggestion was intended to make it unnecessary to cache two variables.  but we can leave it as-is too.
```suggestion
  dockerPromise ??= fs.access('/.dockerenv').then(
    () => true,
    () => false
  );
  const  isDocker = await dockerPromise;
```

Reply:
Sounds good, made the change.